File: 2501.py
def main(n, k):
    t_list = []
    if(k == 0 or k > n) :
        return
    for t in range(1, n+1):
        if (n%t == 0):
            t_list.append(t)
    if (len(t_list) < k) :
        return 0
    return t_list[k-1]

n, k = list(map(int, input().split()))
print(main(n, k))

File: algorithm/src/2501.kt
fun main() {
    val (n, k) = readLine()!!.split(" ").map { it.toInt() }
    val result = findKthFactor(n, k)
    println(result)
}

fun findKthFactor(n: Int, k: Int): Int {
    val tList = mutableListOf<Int>()
    if (k == 0 || k > n) {
        return 0
    }
    for (t in 1..n) {
        if (n % t == 0) {
            tList.add(t)
        }
    }
    if (tList.size < k) {
        return 0
    }
    return tList[k - 1]
}
